library/std/src/sys/solid/fs.rs

```diff
@@ -325,14 +378,17 @@ fn cstr(path: &Path) -> io::Result<CString> {
 
 impl File {
     pub fn open(path: &Path, opts: &OpenOptions) -> io::Result<File> {
```

Comment:
There seems to be some inconsistency between the different targets in offering different combinations of `open`, `open_c` and `open_native`. Could this all be cleaned up to have a single `open` method that takes a `NativePath`?

Reply:
I *think* my intention was to avoid touching as much code as possible and some parts are using `Path` quite a bit so it's a bit more involved to change them to use `NativePath` or adding in the conversions. But I can take another look.